src/Parsers/formatAST.cpp

```diff
@@ -4,11 +4,11 @@
 namespace DB
 {
 
-void formatAST(const IAST & ast, WriteBuffer & buf, bool hilite, bool one_line)
+void formatAST(const IAST & ast, WriteBuffer & buf, bool hilite, bool one_line, bool show_secrets)
 {
     IAST::FormatSettings settings(buf, one_line);
     settings.hilite = hilite;
-
+    settings.show_secrets = show_secrets;
```

Comment:
Since you've added `show_secrets` as a c-tor parameter, it's better to initialize it there.